stl/src/atomic.cpp

```diff
@@ -5,32 +5,21 @@
 
 #include <yvals.h>
 
-#include <intrin.h>
-#pragma warning(disable : 4793)
+#include <synchapi.h>
 
-_EXTERN_C
+namespace {
+    // MUTEX FOR shared_ptr ATOMIC OPERATIONS
+    SRWLOCK _Shared_ptr_lock = SRWLOCK_INIT;
+} // unnamed namespace
 
-// SPIN LOCK FOR shared_ptr ATOMIC OPERATIONS
-volatile long _Shared_ptr_flag;
+_EXTERN_C
 
 _CRTIMP2_PURE void __cdecl _Lock_shared_ptr_spin_lock() { // spin until _Shared_ptr_flag successfully set

```

Comment:
This comment refers to a variable that no longer exists. I recommend replacing it with a comment about the name of the function:
```suggestion
_CRTIMP2_PURE void __cdecl _Lock_shared_ptr_spin_lock() { // TRANSITION, ABI: "spin_lock" name is outdated
```